File: mobile/android/app/src/main/java/com/altonmobile/telegram/TelegramLoginModule.kt
package com.altonmobile.telegram

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

/**
 * Bridge for Telegram Login SDK. Wire org.telegram:login-sdk when GitHub Packages
 * credentials are available — see mobile/README.md.
 */
class TelegramLoginModule(reactContext: ReactApplicationContext) :
  ReactContextBaseJavaModule(reactContext) {

  override fun getName(): String = "TelegramLoginModule"

  @ReactMethod
  fun login(promise: Promise) {
    promise.reject(
      "E_TELEGRAM_SDK",
      "Add org.telegram:login-sdk (GitHub Packages) and TelegramLogin.init — see mobile/README.md"
    )
  }
}

File: mobile/ios/AltonMobile/TelegramLoginModule.m
#import <React/RCTBridgeModule.h>

@interface TelegramLoginModule : NSObject <RCTBridgeModule>
@end

@implementation TelegramLoginModule

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(login:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
{
  reject(@"E_TELEGRAM_SDK", @"Integrate Telegram Login iOS SDK (SPM) and return id_token — see mobile/README.md. Backend: POST /api/login/telegram-sdk.", nil);
}

@end
